### DISCUSSION TITLE
Lint and Format does not exclude the generated source codes

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#257] [Q&A] [answered]
I follow the getting started guide to setup DataFrame in my Spring/Kotlin project.
Kotlin 1.8.10, DataFrame 0.9.1, and I added ksp plugin manually due to there is a warning #256  when building the project.

When I build the project, it seems the SourceSets excludes does not work.

```bash
> Task :formatKotlinMain
Execution optimizations have been disabled for task ':formatKotlinMain' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'D:\FreelanceProjects\sami\etip-backend\build\generated\ksp\main\kotlin'. Reason: Task ':formatKotlinMain' uses this output o
f task ':kspKotlin' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refe
r to https://docs.gradle.org/7.6/userguide/validation_problems.html#implicit_dependency for more details about this problem.

````

**Comment by hantsy** [comment]
Remove the ksp plugin, the format/link works, but #256 is another problem

**Comment by Jolanrensen** [comment, marked as answer]
Hi,
Indeed, we need to update to 1.8.10 (and eventually 1.8.20). Since we use KSP, your Kotlin version needs to be an exact match with the project. So, if you can, stay on Kotlin 1.8.0 until our next update. Otherwise, you can try to build the project from sources, change the kotlin version and install to maven local (or make a PR and use a dev-version after it's been merged)
